crates/agglayer-gcp-kms/src/lib.rs

```diff
@@ -25,6 +37,52 @@ pub struct KMS {
     config: GcpKmsConfig,
 }
 
+// Helper function to get a value from environment variables or configuration
+// with fallbacks.
+// The function returns the value in the following order of precedence:
+// 1. The primary environment variable.
+// 2. The fallback environment variable (if provided).
+// 3. The primary configuration value.
+// 4. The fallback configuration value (if provided).
+// If none are found, it returns an error.
+fn from_env_or_conf<T>(
+    env_key: &str,
+    env_key_fallback: Option<&str>,
+    config_value: &Option<T>,
+    config_value_fallback: &Option<T>,
```

Comment:
General nit: Passing `Option<&T>` is usually preferable to `&Option<T>`. It is a bit more flexible. The caller can easily use `.as_ref()` to get `Option<&T>` but going the other way around is not always easy.